app/controllers/incomeSources/add/AddIncomeSourceStartDateCheckController.scala

```diff
@@ -0,0 +1,288 @@
+          implicit mtdItUser =>
+            handleRequest(
+              isAgent = true,
+              incomeSourceType = incomeSourceType
+            )
+        }
+  }
+
+  private def submitAgent(incomeSourceType: IncomeSourceType): Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService) flatMap {
+          implicit mtdItUser =>
+            handleSubmitRequest(
+              isAgent = true,
+              incomeSourceType = incomeSourceType
+            )
+        }
+  }
+
+  def handleRequest(isAgent: Boolean, incomeSourceType: IncomeSourceType)
+                   (implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
+
+    if(isEnabled(IncomeSources)) {
+      (getStartDate(incomeSourceType), getCalls(isAgent, incomeSourceType)) match {
+        case (None, _) =>
+          Logger("application")
+            .error(s"[AddIncomeSourceStartDateCheckController][handleRequest]: " +
+              s"failed to get start date for $incomeSourceType from session")
+          Future(
+            getErrorHandler(isAgent).showInternalServerError()
+          )
+        case (Some(startDate), (backCall, postAction, _)) =>
+          Future(
+            Ok(
+              addIncomeSourceStartDateCheckView(
+                form = form(incomeSourceType.addIncomeSourceStartDateCheckMessagesPrefix),
+                postAction = postAction,
+                backUrl = backCall.url,
+                isAgent = isAgent,
+                incomeSourceStartDate = longDate(startDate.toLocalDate).toLongDate
+              )
+            )
+          )
+      }
+    } else {
+      Future(
+        Ok(
+          customNotFoundErrorView()
+        )
+      )
+    }
+  } recover {
+    case ex: Exception =>
+      Logger("application").error(s"[AddIncomeSourceStartDateCheckController][handleRequest]: " +
+        s"Error getting AddIncomeSourceStartDateCheck page: ${ex.getMessage}")
+      getErrorHandler(isAgent).showInternalServerError()
+  }
+
+  def handleSubmitRequest(isAgent: Boolean,
+                          incomeSourceType: IncomeSourceType)
+                         (implicit mtdItUser: MtdItUser[_]): Future[Result] = {
+
+    if (isDisabled(IncomeSources)) {
+      Future(
+        Ok(
+          customNotFoundErrorView()
+        )
+      )
+    } else {
+      (getStartDate(incomeSourceType), getCalls(isAgent, incomeSourceType)) match {
+        case (None, _) =>
+          Logger("application").error(s"[AddIncomeSourceStartDateCheckController][handleSubmitRequest]: " +
+            s"failed to get start date from session for incomeSourceType: $incomeSourceType")
+          Future(
+            getErrorHandler(isAgent).showInternalServerError()
+          )
+        case (Some(startDate), (backCall, postAction, successCall)) =>
+
+          form(incomeSourceType.addIncomeSourceStartDateCheckMessagesPrefix)
```

Comment:
if we can move this specific code, related to form creation into separate method ?